src/mds/FSMap.cc

```diff
@@ -760,18 +746,25 @@ std::map<mds_gid_t, MDSMap::mds_info_t> FSMap::get_mds_info() const
   return result;
 }
 
-const MDSMap::mds_info_t* FSMap::get_available_standby(fs_cluster_id_t fscid) const
+const MDSMap::mds_info_t* FSMap::get_available_standby(const Filesystem& fs) const
 {
+  const bool upgradeable = fs.is_upgradeable();
   const mds_info_t* who = nullptr;
   for (const auto& [gid, info] : standby_daemons) {
     ceph_assert(info.rank == MDS_RANK_NONE);
     ceph_assert(info.state == MDSMap::STATE_STANDBY);
 
     if (info.laggy() || info.is_frozen()) {
       continue;
+    } else if (!info.compat.writeable(fs.mds_map.compat)) {
+      /* standby is not compatible with this fs */
+      continue;
+    } else if (!upgradeable && !fs.mds_map.compat.writeable(info.compat)) {
```

Comment:
This doesn't seem correct; an MDS might support features which the existing FS can't handle, but that doesn't mean the MDS can't write to the existing FS while using an older mode, does it?

Reply:
In theory but the MDS is not coded to handle that. snapv2 comes to mind, again.

Reply:
Should this be, !upgradeable || !fs.mds_map.compat.writeable(info.compat) ?

Reply:
No. If the file system is not upgradeable but `fs.mds_map.compat.writeable(info.compat) == true`, we still want to be able to promote that standby.